src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -85,7 +86,8 @@ case class AppDefinition(
   def portIndicesAreValid(): Boolean = {
     val validPortIndices = 0 until hostPorts.size
     healthChecks.forall { hc =>
-      validPortIndices contains hc.portIndex
+      (hc.protocol == Protocol.COMMAND && hc.portIndex == 0) ||
```

Comment:
Easy mistake to make -- there are a lot of ports-related knobs to tweak.  The PORT* environment variables are the ports on the slave that have been allocated to the task by Mesos via the resource offer mechanism.